File: clients/android/NewsBlur/src/com/newsblur/util/NotifyDismissReceiver.java
package com.newsblur.util;

import com.newsblur.activity.Reading;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.AsyncTask;

public class NotifyDismissReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(final Context c, final Intent i) {
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... arg) {
                String storyHash = i.getStringExtra(Reading.EXTRA_STORY_HASH);
                FeedUtils.offerInitContext(c);
                FeedUtils.dbHelper.putStoryDismissed(storyHash);
                return null;
            }
        }.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
    }

}

File: clients/android/NewsBlur/src/com/newsblur/util/NotifyDismissReceiver.kt
package com.newsblur.util

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.newsblur.activity.Reading
import com.newsblur.database.BlurDatabaseHelper
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

@AndroidEntryPoint
class NotifyDismissReceiver : BroadcastReceiver() {

    @Inject
    lateinit var dbHelper: BlurDatabaseHelper

    override fun onReceive(c: Context, i: Intent) {
        val storyHash = i.getStringExtra(Reading.EXTRA_STORY_HASH)
        NBScope.executeAsyncTask(
                doInBackground = {
                    dbHelper.putStoryDismissed(storyHash)
                }
        )
    }
}